openhantek/src/docks/VoltageDock.cpp

```diff
@@ -129,3 +154,38 @@ void VoltageDock::setUsed(ChannelID channel, bool used) {
     QSignalBlocker blocker(channelBlocks[channel].usedCheckBox);
     channelBlocks[channel].usedCheckBox->setChecked(used);
 }
+
+void VoltageDock::setProbeGain(ChannelID channel, unsigned int probeIndex) {
+    if (channel < 0 ||channel >= spec->channels) return ;
+    if(probeIndex > spec->channels) return;
+
+    QSignalBlocker blocker(channelBlocks[channel].probeGainCombobox);
+
+    channelBlocks[channel].probeGainCombobox->setCurrentIndex(probeIndex);
+
+}
+
+/// \brief Update the combobox with the gain values for the probe
+void VoltageDock::probeGainSettingsUpdated() {
+
+    for(unsigned int channel = 0; channel < scope->voltage.size(); channel++) {
+        if(channel < spec->channels) {
+            QSignalBlocker blocker(channelBlocks[channel].probeGainCombobox);
+            //Remove all the old values
+            channelBlocks[channel].probeGainCombobox->clear();
+            // Rebuild the combobox with the new values
+            QStringList probeGainStrings;
+            for(double probe_gain: scope->voltage[channel].probeGainSteps)
+                    probeGainStrings << valueToString(probe_gain, UNIT_TIMES, 0);
+            channelBlocks[channel].probeGainCombobox->addItems(probeGainStrings);
+            // If the index of the list is outside the new list, set index to 0 and notify others
+            if(scope->voltage[channel].probeStepIndex >= scope->voltage[channel].probeGainSteps.size()){
+                this->scope->voltage[channel].probeStepIndex = 0;
+                emit probeGainChanged(channel, 0);
+
+            }
```

Comment:
Have you applied the clang-format tool? In QtCreator you can right click the "format" target in the projects tree view and click "execute". Be careful, this will format all the code (and might add format changes to files you haven't touched). You can also execute Clang-Format from the Tools menu.

Reply:
I do use Clion to write my code. Normally I write python and i'n a PEP8 nazi :P. For c++ I didn't know the convention of the project.